code/game/objects/items/devices/data_detector.dm

```diff
@@ -43,22 +42,19 @@
 		var/detected
 		for(var/DT in objects_to_detect)
 			if(istype(I, DT))
-				if(!detect_empty_vial_boxes && istype(I, /obj/item/storage/fancy/vials/random))
-					if(!I.contents)
-						continue
-				detected = TRUE
+				if((istype(I, /obj/item/storage/fancy/vials/random) && !I.contents.len) || (istype(I, /obj/item/reagent_container/glass/beaker/vial/random) && !I.reagents.total_volume))
+					break //We don't need to ping already looted containers
+				else
+					detected = TRUE
 			if(I.contents)
 				for(var/obj/item/CI in I.contents)
 					if(istype(CI, DT))
-						if(!detect_empty_vial_boxes && istype(I, /obj/item/storage/fancy/vials/random))
-							if(!I.contents)
-								continue
-						detected = TRUE
-						break
+						if((istype(I, /obj/item/storage/fancy/vials/random) && !I.contents.len) || (istype(I, /obj/item/reagent_container/glass/beaker/vial/random) && !I.reagents.total_volume))
```

Comment:
uh shouldn't that be CI ?

Reply:
Yes, yes it should be. Not sure how I missed that.